lib/array.js

```diff
@@ -181,21 +181,31 @@ module.exports = (api) => {
     //   result - array result
   ) => {
     const len = items.length;
-    let i = 0;
     done = api.metasync.cb(done);
+    if (!len) return done;
 
-    function next() {
-      if (i === len) return done();
-      fn(items[i], (err, accepted) => {
-        if (err) return done(err);
-        if (accepted) return done(items[i]);
-        i++;
-        next();
-      });
-    }
+    let count = 0;
+    const output = new Array(len);
+    let finished = false;
+    const finish = () => {
+      finished = true;
+      done(null, [items[output.findIndex(element => element)]]);
+    };
 
-    if (len > 0) next();
-    else done();
+    items.forEach((item, index) => {
+      fn(item, (err, accepted) => {
+        if (err instanceof Error) {
+          if (!finished) {
```

Comment:
This callback should be probably changed to check for this condition on start, to return from all callbacks immediately when there is error in one of them.